Fix Meson error when test dependencies are missing.

Based on this upstream commit, with some adjustments to apply on 2.8.0:
https://gitlab.freedesktop.org/spice/libcacard/-/commit/cf6121deb47e437aa566cfeb29a03da93c432394

diff --git a/tests/meson.build b/tests/meson.build
index 87945a2..baefaf5 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -9,15 +9,6 @@ modutil_dep = find_program('modutil', required: false)
 openssl_dep = find_program('openssl', required: false)
 softhsm_util_dep = find_program('softhsm2-util', required: false)
 
-if pkcs11_tool_dep.found() and p11tool_dep.found() and modutil_dep.found() and openssl_dep.found() and softhsm_util_dep.found()
-  softhsm = custom_target(
-    'softhsm2.conf',
-    command: find_program('setup-softhsm2.sh'),
-    capture: true,
-    output: 'setup-softhsm2.log',
-  )
-endif
-
 libcacard_test = executable(
   'libcacard',
   ['libcacard.c', 'common.c'],
@@ -64,9 +55,18 @@ hwtests_test = executable(
   dependencies: [libcacard_dep],
 )
 
-test(
-  'hwtests',
-  hwtests_test,
-  depends: [softhsm],
-  env: env,
-)
+if pkcs11_tool_dep.found() and p11tool_dep.found() and modutil_dep.found() and openssl_dep.found() and softhsm_util_dep.found()
+  softhsm = custom_target(
+    'softhsm2.conf',
+    command: find_program('setup-softhsm2.sh'),
+    capture: true,
+    output: 'setup-softhsm2.log',
+  )
+
+  test(
+    'hwtests',
+    hwtests_test,
+    depends: [softhsm],
+    env: env,
+  )
+endif
